packages/yourloops/components/statistics/use-patient-statistics.hook.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+import { type BgPrefs, CBGStatType } from 'dumb'
+import { type BgType, type CbgRangeStatistics, type DateFilter, DatumType, type MedicalData, TimeService } from 'medical-domain'
+import { GlycemiaStatisticsService } from 'medical-domain/dist/src/domains/repositories/statistics/glycemia-statistics.service'
+
+interface usePatientStatisticsProps {
+  medicalData: MedicalData
+  bgPrefs: BgPrefs
+  dateFilter: DateFilter
+  bgSource: BgType
+}
+
+export interface SensorUsageData {
+  total: number
+  usage: number
+}
+
+interface usePatientStatisticsReturn {
+  cbgStatType: CBGStatType
+  numberOfDays: number
+  cbgPercentageBarChartData: CbgRangeStatistics
+  sensorUsageData: SensorUsageData
+  cbgSelected: boolean
+}
+
+export const usePatientStatistics = ({ medicalData, bgPrefs, bgSource, dateFilter }: usePatientStatisticsProps): usePatientStatisticsReturn => {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏No state management. Nothing worth being in a custom hook. Why does this exists? I imagine that this was done to avoid code duplication, right? But this does not need to be in a hook, it could just be a normal function.
I also question the code duplication that this avoids. Because when `packages/blip/app/components/chart/deviceUsage.js` is going to be removed, we will forget about this weird function/hook and will end up keeping it.